v2/auth/cachedauth/sign_in.go

```diff
@@ -0,0 +1,61 @@
+package cachedauth
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/SKF/go-utility/v2/auth"
+)
+
+var lock sync.Mutex
+var tokens auth.Tokens
+
+var config *Config
+
+// Config is the configuration of the package
+type Config struct {
+	Stage string
+}
+
+// Configure will configure the package
+func Configure(conf Config) {
+	lock.Lock()
```

Comment:
Why do we need the mutex here?

Reply:
It would'nt be very nice with  a mixed configuration state between this package and the regular auth.